### ISSUE TITLE
Errors on AWS with 8 NVIDIA K80 cards.

### ISSUE THREAD

**Issue by Hui-Zhi** [tensorflow/tensorflow#5934]
### What related GitHub issues or StackOverflow threads have you found by searching the web for your problem?

convnet-benchmarks/tensorflow$ python benchmark_alexnet.py 
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcublas.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcudnn.so.5 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcufft.so.8.0 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcurand.so.8.0 locally
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:925] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 0 with properties: 
name: Tesla K80
major: 3 minor: 7 memoryClockRate (GHz) 0.8235
pciBusID 0000:00:17.0
Total memory: 11.17GiB
Free memory: 11.11GiB
W tensorflow/stream_executor/cuda/cuda_driver.cc:572] creating context when one is currently active; existing: 0x31a4790
I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:925] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 1 with properties: 

I tensorflow/core/common_runtime/gpu/gpu_device.cc:972] DMA: 0 1 2 3 4 5 6 7 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 0:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 1:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 2:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 3:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 4:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 5:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 6:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 7:   Y Y Y Y Y Y Y Y 
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla K80, pci bus id: 0000:00:17.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:1) -> (device: 1, name: Tesla K80, pci bus id: 0000:00:18.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:2) -> (device: 2, name: Tesla K80, pci bus id: 0000:00:19.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:3) -> (device: 3, name: Tesla K80, pci bus id: 0000:00:1a.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:4) -> (device: 4, name: Tesla K80, pci bus id: 0000:00:1b.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:5) -> (device: 5, name: Tesla K80, pci bus id: 0000:00:1c.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:6) -> (device: 6, name: Tesla K80, pci bus id: 0000:00:1d.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:7) -> (device: 7, name: Tesla K80, pci bus id: 0000:00:1e.0)
E tensorflow/stream_executor/cuda/cuda_driver.cc:1110] failed to synchronize the stop event: CUDA_ERROR_MISALIGNED_ADDRESS
E tensorflow/stream_executor/cuda/cuda_timer.cc:54] Internal: error destroying CUDA event in context 0x34d6b30: CUDA_ERROR_MISALIGNED_ADDRESS
E tensorflow/stream_executor/cuda/cuda_timer.cc:59] Internal: error destroying CUDA event in context 0x34d6b30: CUDA_ERROR_MISALIGNED_ADDRESS
F tensorflow/stream_executor/cuda/cuda_dnn.cc:1979] failed to enqueue convolution on stream: CUDNN_STATUS_EXECUTION_FAILED
Aborted (core dumped)

### Environment info
Operating System:

DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.1 LTS"

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):
CUDA 8.0 (from: http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-repo-ubuntu1604_8.0.44-1_amd64.deb)
cudnn:5.1

If installed from source, provide 
1. https://github.com/tensorflow/tensorflow/archive/v0.11.0.tar.gz

2. The output of `bazel version`
Build label: 0.3.2
Build target: bazel-out/local-fastbuild/bin/src/main/java/com/google/devtools/build/lib/bazel/BazelServer_deploy.jar
Build time: Fri Oct 7 17:25:10 2016 (1475861110)
Build timestamp: 1475861110
Build timestamp as int: 1475861110

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)
python https://github.com/soumith/convnet-benchmarks/blob/master/tensorflow/benchmark_alexnet.py

### What other attempted solutions have you tried?
reinstall tensorflow and reconfigure the environment.
If only has 1 K80 card, it is success.

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

**Comment by poxvoculi** [comment]
Were you able to get any more of a stack trace at the failure?  I haven't seen this CUDA_ERROR_MISALIGNED_ADDRESS error before, and I'm wondering where the program is executing when it arises.  You say the error does not arise using only one k80.  What if you use only 2?

**Comment by Hui-Zhi** [comment]
@poxvoculi I am sorry that on AWS I am only able to use either 8 or 1 GPU on a instance. and on K520 there's no such problem.
Regarding the stack trace, I have already delete the instance because it is very expensive. What stack trace do you need and how can I get it? I am new to tensorflow.

**Comment by poxvoculi** [comment]
@zheng-xq what do you think?

**Comment by gunan** [comment]
@Hui-Zhi you can set the environment variable `CUDA_VISIBLE_DEVICES` to set which GPUs are available for tensorflow to use. For example if you set `CUDA_VISIBLE_DEVICES=0,1` in your environment before you run your code, only two GPUs will be used by TF.

**Comment by Hui-Zhi** [comment]
@gunan No, I did not.

**Comment by aselle** [comment]
@tfboyd, can you take a look at this?

**Comment by tfboyd** [comment]
This issue is a little stale, but I should be on AWS this week and it should be easy enough to run the code from:

`python https://github.com/soumith/convnet-benchmarks/blob/master/tensorflow/benchmark_alexnet.py`

I will be using TF 1.0 so it will not be apples to apples but knowing it runs fine on TF 1.0 will be nice.

**Comment by tfboyd** [comment]
I tested this on a p2.8xLarge with TF 1.0 (hash:c7fca9e) as follows:

`python benchmark_alexnet.py`
and
`CUDA_VISIBLE_DEVICES=0 python benchmark_alexnet.py`

It worked fine for both cases.  Unless I am reading the code incorrectly (which happens), this [benchmark](https://github.com/soumith/convnet-benchmarks/blob/master/tensorflow/benchmark_alexnet.py) is not multi-GPU so the result is testing single GPU performance for AlexNet.  My results just because I have them:

K80
2017-02-21 22:05:39.664444: Forward across 100 steps, 0.064 +/- 0.006 sec / batch
2017-02-21 22:06:01.289066: Forward-backward across 100 steps, 0.186 +/- 0.019 sec / batch

Local GTX 1080 (also my display GPU)
2017-02-21 14:24:55.096074: Forward across 100 steps, 0.023 +/- 0.003 sec / batch
2017-02-21 14:25:03.292397: Forward-backward across 100 steps, 0.070 +/- 0.007 sec / batch

p.s. I did have to install future as well `sudo pip install future`
